File: June/08-06-2024/Index of an Extra Element.java
//{ Driver Code Starts
import java.util.*;

class ExtraElement {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();
        while (t-- > 0) {
            int n = sc.nextInt();
            int[] a = new int[n];
            int[] b = new int[n - 1];
            for (int i = 0; i < n; i++) a[i] = sc.nextInt();
            for (int i = 0; i < n - 1; i++) b[i] = sc.nextInt();
            Solution g = new Solution();
            System.out.println(g.findExtra(n, a, b));
        }
    }
}
// } Driver Code Ends




/*Complete the function below*/
class Solution {
    public int findExtra(int n, int arr1[], int arr2[]) {
        // Your Code Start
           int left = 0, right = n - 1;

        while (left < right) {
            int mid = left + (right - left) / 2;
            if (arr1[mid] == arr2[mid]) {
                left = mid + 1;
            } else {
                right = mid;
            }
        }

        return left;
      //Your Code End
    }
}

File: June/08-06-2024/Index of an Extra Element.py
class Solution:
    def findExtra(self,n,a,b):
        # Your Code Start
        left, right = 0, n - 1
        
        while left < right:
            mid = left + (right - left) // 2
            if a[mid] == b[mid]:
                left = mid + 1
            else:
                right = mid
        
        return left
    #Your Code End




#{ 
 # Driver Code Starts
if __name__ == "__main__":
    t = int(input())
    for i in range(t):
        n = int(input())
        a = list(map(int, input().split()))
        b = list(map(int, input().split()))
        print(Solution().findExtra(n, a, b))

# } Driver Code Ends
